proposals/lip-0058.md

```diff
@@ -488,6 +488,13 @@ The function is called in the after application processing stage of the genesis
 def setBFTParameters(precommitThreshold, certificateThreshold, validatorList):
     if length(validatorList) > LSK_BFT_BATCH_SIZE:
         raise Exception("Given list of validators is larger than the maximum allowed value.")
+    if len(validatorList) != len(set([validator.address for validator in validatorList])):
+        raise Exception("Duplicate validator address in validator list.")
+    if len(validatorList) != len(set([validator.generatorKey for validator in validatorList])):
```

Comment:
I would remove this check. In the application layer, including validators module and PoA/PoS module, we do allow that two validator accounts use the same generator key. Therefore, it could happen that two validator accounts are created with the same generator key, and eventually both become active validators. But when `setBFTParameters` is called during the processing of last block of the round, then `setBFTParameters` throws and the block is invalid although there was in fact no problem with the block. That means, the chain is stuck from there on.

Reply:
Okay, sounds good to me not require generator key uniqueness.

Reply:
ah i didn't know this is allowed, but looks good to me.